File: backend/src/main/java/com/example/Flicktionary/domain/genre/entity/Genre.java
package com.example.Flicktionary.domain.genre.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Genre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    public Genre(String name) {
        this.name = name;
    }
}

File: backend/src/main/java/com/example/Flicktionary/domain/genre/entity/Genre.kt
package com.example.Flicktionary.domain.genre.entity

import jakarta.persistence.*

@Entity
class Genre(
    @Column(nullable = false)
    val name: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private var _id: Long? = null

    var id: Long
        get() = _id ?: 0
        set(value) {
            _id = value
        }
}